contrib/mpiexecjl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env julia
+# -*- julia -*-
+#
+# Copyright (C) 2020 Mosè Giordano
+# License is MIT "Expat"
+#
+### Commentary:
+#
+# Command line utility to call the `mpiexec` binary used by the `MPI.jl` version
+# in the given Julia project.  It has the same syntax as the `mpiexec` binary
+# that would be called, with the additional `--project=...` to select a
+# different Julia project.
+#
+# Examples of usage (the MPI flags available depend on the MPI implementation
+# called):
+#
+#   $ mpiexecjl --version
+#   $ mpiexecjl -n 40 mpi-script.jl
+#   $ mpiexecjl --project=my_experiment -n 80 --oversubscribe mpi-script.jl
+#
+### Code:
+
+# From BinaryBuilder.jl
+function extract_flag!(ARGS, flag, val = "")
+    for f in ARGS
+        if f == flag || startswith(f, string(flag, "="))
+            # Check if it's just `--flag` or if it's `--flag=foo`
+            if f != flag
+                val = split(f, '=')[2]
+            end
+
+            # Drop this value from our ARGS
+            filter!(x -> x != f, ARGS)
+            return (true, String(val))
+        end
+    end
+    return (false, val)
+end
+
+_ARGS = deepcopy(ARGS)
+project, project_dir = extract_flag!(_ARGS, "--project")
+
+const script = """
+    using MPI
+    ENV["JULIA_PROJECT"] = dirname(Base.active_project())
+    mpiexec() do exe
+        if exe isa Cmd
+            cmd = Cmd([MPI.mpiexec_path, ARGS...])
+        else # if exe isa String
+            cmd = Cmd([exe, ARGS...])
+        end
+        run(cmd)
```

Comment:
Any reason why not just do
```suggestion
        run(`$exe $ARGS`)
```
?

Reply:
Not at computer now, but if I remember correctly this doesn't work when `exe` is a `Cmd`, which is the case when `JULIA_MPIEXEC_BINARY=system`

Reply:
It does for me:
```
julia> exe = `mpiexec -foo`; ARGS = ["aa", "bb"]
2-element Array{String,1}:
 "aa"
 "bb"

julia> `$exe $ARGS`
`mpiexec -foo aa bb`
```

Reply:
Well, at this point I don't know what I had tried :confused:  I simplified the function as you suggested